packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -1449,6 +1451,46 @@ void main() {
     expect(tester.getTopRight(find.text('text')).dx, lessThanOrEqualTo(tester.getTopLeft(find.byIcon(Icons.satellite)).dx));
   });
 
+  testWidgets('InputDecorator disabledColor should be set', (WidgetTester tester) async {
+    const Color expectedColor = Color(0xFF123456);
+
+    await tester.pumpWidget(
+      buildInputDecorator(
+        disabledColor: expectedColor,
+        isFocused: false,
+        decoration: const InputDecoration(
+          icon: Icon(Icons.add_circle),
+          enabled: false,
+        )
+      )
+    );
+
+    final IconTheme iconTheme = tester.firstWidget(
```

Comment:
A possibly cleaner way of getting the theme would be: 

```dart
    final Element icon = tester.element(find.byIcon(Icons.add_circle));
    final IconTheme iconTheme = icon.findAncestorWidgetOfExactType<IconTheme>();
```

This could apply to the other test below as well.

Reply:
Thanks for that, I changed the behaviour in all tests and added some other tests.